src/MemoryTracing/MemoryTracingIntegrationTest.cpp

```diff
@@ -28,31 +34,37 @@ using orbit_memory_tracing::MemoryInfoProducer;
 
 class BufferMemoryInfoListener : public MemoryInfoListener {
  public:
+  explicit BufferMemoryInfoListener(uint64_t sampling_start_timestamp_ns,
+                                    uint64_t sampling_period_ns, bool enable_cgroup_memory)
+      : sampling_start_timestamp_ns_(sampling_start_timestamp_ns),
+        sampling_period_ns_(sampling_period_ns),
+        enable_cgroup_memory_(enable_cgroup_memory) {}
+
   void OnCGroupMemoryUsage(orbit_grpc_protos::CGroupMemoryUsage cgroup_memory_usage) override {
-    ProducerCaptureEvent event;
-    *event.mutable_cgroup_memory_usage() = std::move(cgroup_memory_usage);
-    {
-      absl::MutexLock lock{&events_mutex_};
-      events_.emplace_back(std::move(event));
-    }
+    uint64_t sampling_window_id = GetSamplingWindowId(cgroup_memory_usage.timestamp_ns());
+
+    absl::MutexLock lock(&in_progress_wrappers_mutex_);
+    *in_progress_wrappers_[sampling_window_id].mutable_cgroup_memory_usage() =
+        std::move(cgroup_memory_usage);
+    ProcessMemoryEventWrapperIfReady(sampling_window_id);
```

Comment:
This seems to be the same code as in `MemoryInfoHandler`. Can't we inherit from `MemoryInfoHandler` and re-use a bunch of this code - or move the shared code into `MemoryInfoListener`? To me it looks like the only difference is what we do with events once they are created.

Reply:
Thanks for the suggestion. As we discussed offline, this part has been moved to `MemoryInfoListener`.